infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -303,26 +333,40 @@ public void read(CorfuPayloadMsg<ReadRequest> msg, ChannelHandlerContext ctx, IS
     private void getKnownAddressesInRange(CorfuPayloadMsg<KnownAddressRequest> msg,
                                           ChannelHandlerContext ctx, IServerRouter r) {
 
-        KnownAddressRequest request = msg.getPayload();
+        Lock lock = resetLock.readLock();
+        lock.lock();
+
         try {
-            Set<Long> knownAddresses = streamLog
-                    .getKnownAddressesInRange(request.getStartRange(), request.getEndRange());
-            r.sendResponse(ctx, msg,
-                    CorfuMsgType.KNOWN_ADDRESS_RESPONSE.payloadMsg(knownAddresses));
-        } catch (DataCorruptionException dc) {
-            r.sendResponse(ctx, msg, CorfuMsgType.ERROR_DATA_CORRUPTION
-                    .payloadMsg(Address.NON_ADDRESS));
-            return;
-        } catch (Exception e) {
-            handleException(e, ctx, msg, r);
+            KnownAddressRequest request = msg.getPayload();
+            try {
```

Comment:
Epoch check after lock.lock().

Reply:
I have changed `processRequest` could you please take a look

Reply:
@xnull Is epoch check missing here?

Reply:
As discussed twice the check will crash most of the unit tests, here is the build with the change https://travis-ci.org/github/CorfuDB/CorfuDB/builds/716015026